### PR TITLE
Comprehensive tests for sorting behaviour

### PR DESCRIPTION
**Pull request by dorkydhruv**
tests added - 
- test_bundle_priority_calculation_with_cu_limits
- test_nonzero_cu_price_beats_zero_price_with_limit
- test_zero_price_two_signers_can_outrank_low_price
- test_bundle_priority_calculation_with_tips
- test_tip_bundle_outranks_low_priority_fee_bundle

### CODE REVIEW

**Comment by dorkydhruv** [comment]
@OliverNChalk do we need more tests here, or is this sufficient enough? If yes, any edge cases we might want to check?

File: core/src/bundle_stage/bundle_storage.rs [hunk 1]
```diff
@@ -647,7 +641,416 @@ mod tests {
...
+            ];
+            let msg = Message::new(&ixs, Some(&payer.pubkey()));
+            Transaction::new(&[&payer], msg, blockhash)
+        };
+        let tx_tip = solana_system_transaction::transfer(
+            &tip_funder,
+            &tip_account.pubkey(),
+            50_000,
+            blockhash,
+        );
```
Comment (Art3miX) [inline core/src/bundle_stage/bundle_storage.rs:1004]: I wonder, should we setup all TXs to hold the exact same IXs to test the actual values?
What I mean is that here you have 2 TXs, that 1 hold 3 IXs and 2nd holds 4 IXs, this adds the 4th IX cu usage into calculation.

What if the first TX will also have a tip transfer of 0, this means that in theory both TXs should have the same exact cu usage, and the only changing factor is the amount of rewards that is paid.

This will make it a better tests because we only testing the actual priority calculation by only 1 factor while keeping the CU usage the same (rewards / cu_usage)

**Review by Art3miX — CHANGES_REQUESTED** [review]
It was pretty hard to follow the tests, mainly because even tho they are similar, each test has its own way of doing stuff which is very error prone and hard to review.

The comments are not touching the test themselves or if we have missing tests, rather the way of organizing them so it will be easy to follow.

I will think a little more about what tests are missing if any.

File: core/src/bundle_stage/bundle_storage.rs (hunk 1 above)
Comment (dorkydhruv) [reply core/src/bundle_stage/bundle_storage.rs]: makes sense, we can have tip transactions even if they are zero.
